include/nitro/broken_options/parser.hpp

```diff
@@ -96,11 +109,10 @@ namespace broken_options
         std::string app_name_;
         std::string about_;
 
-        std::map<std::string, broken_options::option> options_;
-        std::map<std::string, broken_options::multi_option> multi_options_;
-        std::map<std::string, broken_options::toggle> toggles_;
+        std::set<std::string> all_argument_names_;
+        std::deque<nitro::broken_options::group> groups_;
```

Comment:
Why is that a deque?

Reply:
I thought this is the best option... :/

Reply:
Let me rephrase that: Why are you using a `deque` instead of a `vector`?

Reply:
I can't use a vector, because the objects in a vector can be relocated and there are references in to the groups_ list :/